### DISCUSSION TITLE
Square's Wire library

### DISCUSSION THREAD

**Discussion by clamothebb** [line/armeria#5982] [Ideas]
Hello,

A while back, I tried to use Armeria with [Wire](https://square.github.io/wire/), though didn't get very far. Wire generates its own versions of classes like io.grpc.Status, that don't implement protobuf's Message class.

I disregarded the idea of using the two together after this, however seeing that Armeria supports scalapb made me wonder. Do you think an attempt at adding support for Wire could be successful, or does scalapb somehow follow protobuf-java's rules in a way that makes it easier to support?

Thanks!

**Comment by jrhee17** [comment]
Hi, I haven't really tried Wire myself. Let me do a quick POC and get back to you
